amy/workshops/mixins.py

```diff
@@ -149,6 +149,27 @@ def active(self):
         return self.state == "p"
 
 
+class StateExtendedMixin(models.Model):
+    """State field with representation of 'Withdrawn' state, for now only used in
+    TrainingRequest.
+
+    This was rewritten instead of inherited from `StateMixin` - there were some
+    issues with `get_state_display` method for "withdrawn" state."""
```

Comment:
@pbanaszkiewicz could you say more about this issue? I don't really understand why inheriting from this class causes issues?

Reply:
@lauryndbrown I didn't get to the root of the problem. I only discovered that `get_state_display` didn't display correctly the new state - "Withdrawn". Instead there only was `w`, like the field didn't use the new choice values.

If I remember correctly, I rewrote pretty much the whole `StateMixin` except for the `active` method - only it was inherited.